### DISCUSSION TITLE
nx30pro 刷入bl13-uboot无法进入web

### DISCUSSION THREAD

**Discussion by Potterli20** [immortalwrt/immortalwrt#1008] [Q&A] [answered]
同上
![wx_camera_1688831850251.jpg](https://github.com/immortalwrt/immortalwrt/assets/28734349/0f724322-e242-440a-b5d0-886d417bee98)

![wx_camera_1688831748308.jpg](https://github.com/immortalwrt/immortalwrt/assets/28734349/dc5307ae-dc6b-49e1-b05f-fd4d04893fa0)

**Comment by 1715173329** [comment]
openwrt 的 uboot 没有 web 界面，需要自行配置 tftp server。

**Reply by Potterli20** [reply]
问题是我用imm的源去编译，默认不带web？

**Comment by Potterli20** [comment]
> openwrt 的 uboot 没有 web 界面，需要自行配置 tftp server。

请问如何弄，我弄了不生效

**Reply by Potterli20** [reply]
请求失败
![IMG_20230709_011920.jpg](https://github.com/immortalwrt/immortalwrt/assets/28734349/2891d439-765a-48c9-9b7c-b5e7428bd879)

**Reply by Jokerman1991** [reply]
我现在遇到同样的问题，老哥是怎么解决的，实在找不到 tftp 或者 tftpd 工具的详细教程

**Reply by Potterli20** [reply]
> 我现在遇到同样的问题，老哥是怎么解决的，实在找不到 tftp 或者 tftpd 工具的详细教程

你是用tftp刷回去，tftp那里有个日记的，没有找到对印固件他会报错

**Reply by Jokerman1991** [reply]
没办法，拆机了（我的是 wr30u，跟你的其实一个东西，我这个板子连 rx tx gnd 都不标，我是自己试出来的），用ttl读出来，我勒个去：

ARP Retry count exceeded; starting again
Wrong Image Format for bootm command
ERROR: can't get kernel image!
Using ethernet@15100000 device
TFTP from server 192.168.1.254; our IP address is 192.168.1.1
Filename 'immortalwrt-mediatek-filogic-xiaomi_mi-router-wr30u-ubootmod-initramfs-recovery.itb'.
ERROR: reserving fdt memory region failed (addr=4fc00000 size=100000 flags=4)
Load address: 0x46000000

必须用指定的 IP 不说，我之前都试了是 100 以内的，压根没往 254 去想，还有就是 用 tftp 命令还不行，但是用 tftpd 客户端就行，这个客户端设置好 server 目录以后，如果单有线网口且不连 wifi 的情况下（也就是只有一个网卡有效的情况下），设置好 server 目录，client 端也就是路由器那边，自动就识别，真是折腾......

**Reply by Potterli20** [reply]
> 没办法，拆机了（我的是 wr30u，跟你的其实一个东西，我这个板子连 rx tx gnd 都不标，我是自己试出来的），用ttl读出来，我勒个去：
> 
> ARP Retry count exceeded; starting again
> Wrong Image Format for bootm command
> ERROR: can't get kernel image!
> Using ethernet@15100000 device
> TFTP from server 192.168.1.254; our IP address is 192.168.1.1
> Filename 'immortalwrt-mediatek-filogic-xiaomi_mi-router-wr30u-ubootmod-initramfs-recovery.itb'.
> ERROR: reserving fdt memory region failed (addr=4fc00000 size=100000 flags=4)
> Load address: 0x46000000
> 
> 必须用指定的 IP 不说，我之前都试了是 100 以内的，压根没往 254 去想，还有就是 用 tftp 命令还不行，但是用 tftpd 客户端就行，这个客户端设置好 server 目录以后，如果单有线网口且不连 wifi 的情况下（也就是只有一个网卡有效的情况下），设置好 server 目录，client 端也就是路由器那边，自动就识别，真是折腾......

哈哈哈哈，我是用tftp的😂

**Reply by freezeTian** [reply]
我也遇到了同样的问题，TFTP 一点日志都没有，好像TFTP 服务器不存在的样子，你们的路由器还在用吗

**Reply by freezeTian** [reply]
> 没办法，拆机了（我的是 wr30u，跟你的其实一个东西，我这个板子连 rx tx gnd 都不标，我是自己试出来的），用ttl读出来，我勒个去：
> 
> ARP Retry count exceeded; starting again Wrong Image Format for bootm command ERROR: can't get kernel image! Using ethernet@15100000 device TFTP from server 192.168.1.254; our IP address is 192.168.1.1 Filename 'immortalwrt-mediatek-filogic-xiaomi_mi-router-wr30u-ubootmod-initramfs-recovery.itb'. ERROR: reserving fdt memory region failed (addr=4fc00000 size=100000 flags=4) Load address: 0x46000000
> 
> 必须用指定的 IP 不说，我之前都试了是 100 以内的，压根没往 254 去想，还有就是 用 tftp 命令还不行，但是用 tftpd 客户端就行，这个客户端设置好 server 目录以后，如果单有线网口且不连 wifi 的情况下（也就是只有一个网卡有效的情况下），设置好 server 目录，client 端也就是路由器那边，自动就识别，真是折腾......

你的意思是，TFTP 的IP 要指定1.254？

**Comment by Potterli20** [comment, marked as answer]
https://github.com/immortalwrt/immortalwrt/issues/1009

**Comment by hcwhan** [comment]
原来u-boot没web....
我都返厂了 (笑哭)

**Comment by hcwhan** [comment]
@1715173329 你好  请问可以编译兼容恩山论坛u-boot的固件吗 或者在已刷入openwrt可直接web升级的固件

**Reply by Potterli20** [reply]
如果你刷了l大的源，再刷官方op的源，会出bug，会无限重启uboot
建议刷完官方op固件，手动重启（就是拨线）

**Reply by 1715173329** [reply]
我没这个设备，也不用恩山，也没见到恩山 uboot 的源码。
openwrt 的 uboot 也并不差，系统挂了会自己进恢复模式，只不过是目前没有太多详细的说明。
